src/ddscxx/include/org/eclipse/cyclonedds/topic/datatopic.hpp

```diff
@@ -592,9 +592,26 @@ ddsi_serdata * serdata_from_iox_buffer(
 {
   try {
     auto d = new ddscxx_serdata<T>(typecmn, kind);
-    d->iox_chunk = iox_buffer;
-    d->iox_subscriber = sub;
 
+    if (sub == nullptr) {  // serdata from the loaned sample
+#ifndef _WIN32
```

Comment:
`dds/ddsrt/misc.h` has macros for doing this, `DDSRT_WARNING_GNUC_OFF(old-style-cast)` should do this trick (there's a corresponding `ON` macro as well, of course)

Reply:
Thanks for point it :) I was not aware of this